system-design/margo-overview/software-composition.md

```diff
@@ -0,0 +1,285 @@
+        System(cmp, "Component")
+        Rel(appb, apdb, "instantiation")
+        Rel(appb, cmp, "requires")
+    }
+
+    Enterprise_Boundary(dev, "Device") {
+        System(apdd, "ApplicationDeployment", "Application Deployment specification")
+        System(wls, "Workload")
+        Rel(apdd, wls, "configures")
+    }
+
+    BiRel(apdb, apdd, "same")
+    Rel(cmp, wls, "instantiation")
+    UpdateElementStyle(appb, $fontColor="black", $bgColor="blue", $borderColor="grey")
+    UpdateElementStyle(appd, $fontColor="black", $bgColor="blue", $borderColor="grey")
+    UpdateElementStyle(apdb, $fontColor="black", $bgColor="green", $borderColor="grey")
+    UpdateElementStyle(apdd, $fontColor="black", $bgColor="green", $borderColor="grey")
+```
+
+In this stage the [providers][provider-model] are responsible for managing the individual [Workloads][workload].
+
+On a Helm v3 [Deployment Profiles][deployment-profile], a [Workload Fleet Management Agent][wfma] implementation could utilize the Helm API to start the individual Helm Charts.
+
+On a Compose [Deployment Profiles][deployment-profile], a [Workload Fleet Management Agent][wfma] implementation could utilize the Compose CLI to start the individual [Workloads][workload].
+
+The following diagram shows the result of reaching the desired state for an [Application][application] with a Helm v3 [Deployment Profile][deployment-profile] (the result of `helm install`).
+
+```mermaid
+C4Component
+    title Application Deployment: Helm v3 deployment profile
+
+    UpdateLayoutConfig($c4BoundaryInRow="3", $c4ShapeInRow="1")
+
+    System_Boundary(dev1, "Device 1") {
+        System_Boundary(woa1, "Workload Fleet Management Agent") {
+            Component(atb1, "Application Description 1", "ApplicationDescription", "YAML document")
+        }
+
+        System_Boundary(hel1, "Helm") {
+            Component(hc1, "Helm Chart 1")
+            Component(hc2, "Helm Chart 2")
+        }
+
+        System_Boundary(kbl1, "Kubelet") {
+            Component(pod1, "Pod 1")
+            Component(pod2, "Pod 2")
+            Component(pod3, "Pod 3")
+        }
+    }
+
+    Rel(atb1, hc1, "refers")
+    Rel(hc1, pod1, "refers")
+    Rel(atb1, hc2, "refers")
+    Rel(hc2, pod2, "refers")
+    Rel(hc2, pod3, "refers")
+    Rel(hc2, pod3, "refers")
+
+    UpdateElementStyle(atb1, $fontColor="white", $bgColor="blue", $borderColor="grey")
+```
+
+The following diagram shows the result of reaching the desired state for an [Application][application] with a Compose [Deployment Profile][deployment-profile] (the result of the `compose up` CLI call).
+
+```mermaid
+C4Component
+    title Application Deployment: Compose deployment profile
+
+    UpdateLayoutConfig($c4BoundaryInRow="3", $c4ShapeInRow="1")
+
+    System_Boundary(dev1, "Device 1") {
+        System_Boundary(woa1, "Workload Fleet Management Agent") {
+            Component(atb1, "Application Description 1", "ApplicationDescription", "YAML document")
```

Comment:
As mentioned by @phil-abb in [this comment](https://github.com/margo/specification/pull/81#discussion_r2080016984), this should be Application Deployment and not Application Description.

Reply:
@phil-abb could you please check if now it's correct and resolve this conversation, if so? Thanks!